main/com.tunein/device.js

```diff
@@ -0,0 +1,102 @@
+const axios = require('axios');
+const axiosExtensions = require('axios-extensions');
+const url = require('url');
+
+const BASE_URL = 'http://opml.radiotime.com';
+const CONTENT_TYPE = 'application/x-www-form-urlencoded';
+const HTTP_STATUS_OK = 200;
+const SEARCH_PARAM = '/Search.ashx';
+const BROWSE_PARAM = '/Browse.ashx';
+const BROWSE_ITEMS = ['trending', 'music', 'sports', 'talk'];
+
+module.exports = class TuneinRadioDevice extends Tp.BaseDevice {
+    constructor(engine, state, args={}) {
+         super(engine, state);
+         this.uniqueId = 'com.tunein';
+         this.name = 'Tunein';
+         this.description = 'Stream radio through TuneIn API';
+         axios.defaults.headers.post['Content-Type'] = args.content_type || CONTENT_TYPE;
+         axios.defaults.baseURL = args.protocol || BASE_URL;
+         axios.defaults.params = { render: 'json' };
+    }
+
+    _get_tunein_response(req) {
+        return new Promise( (resolve, reject) => {
+            axios.get(req.url, req)
+                .then(response => { 
+                    if (response.data.head.status == HTTP_STATUS_OK) {
+                        let content = response.data.body;
+                        if ('c' in req.params && BROWSE_ITEMS.includes(req.params.c.toLowerCase())) {
+                            return resolve(content);
+                        } else if (content.find(item => item.text.toLowerCase() == 'stations')) {
+                            return resolve(
+                                content.filter(item => item.text.toLowerCase() == 'stations')
+                                       .flatMap(({children}) => children)
+                                       .filter(channel => (channel.type.toLowerCase() == 'audio' && channel.item.toLowerCase() == 'station'))
+                            );
+                        } else {
+                            return resolve(content.filter(channel => (channel.type.toLowerCase() == 'audio' && channel.item.toLowerCase() == 'station')));
+                        }; 
+                    } else {
+                        reject(response.data);
+                        return new Error(`HTTP request error: ${response.data.head.fault}`);
+                    }
+                })
+                .catch(err => throwError("streaming_service_unaccessible"));
+        }).catch(err => throwError("streaming_service_unaccessible"));
+    }
+
+    _get_channel_info(ch_list) {
+        return ch_list.map(
+            (x) => (({text, subtext='', current_track='', now_playing_id='', URL='', formats='', image=''}) => ({text, subtext, current_track, now_playing_id, URL, formats, image}))(x)
+            );
+    }
+
+    async get_search_channel({query, count = 0}) {
+        let req = {};
+        req.params = {};
+        req.url = SEARCH_PARAM;
+        req.params.query = query;
+        const available_channels = await this._get_tunein_response(req);
+        if ([undefined, null, 0].includes(count))
```

Comment:
So, quite precisely, `if (!count)` ???
Why did you go out of your way to spell that out?

(Also, count can never be undefined because you defaulted it in the function signature)